packages/mui-material/src/Grid/Grid.js

```diff
@@ -315,24 +331,44 @@ const Grid = React.forwardRef(function Grid(inProps, ref) {
   // columns set with default breakpoint unit of 12
   const columns = container ? columnsProp || 12 : columnsContext;
 
+  const breakpointsDefaultValues = {};
+  const breakpointsProps = {};
+  const propsRestFiltered = {};
+
+  breakpoints.keys.forEach((key) => {
+    breakpointsDefaultValues[key] = false;
+  });
+
+  Object.keys(propsRest).forEach((key) => {
+    const value = propsRest[key];
+
+    if (key in breakpoints.values) {
+      breakpointsProps[key] = value;
+    } else {
+      propsRestFiltered[key] = value;
+    }
+  });
+
+  const breakpointsValues = {
+    ...breakpointsDefaultValues,
+    ...breakpointsProps,
+  };
+
   const ownerState = {
     ...props,
     columns,
     container,
     direction,
     item,
-    lg,
-    md,
-    sm,
     rowSpacing,
     columnSpacing,
     wrap,
-    xl,
-    xs,
     zeroMinWidth,
+    spacing,
+    ...breakpointsValues,
   };
 
-  const classes = useUtilityClasses(ownerState);
+  const classes = useUtilityClasses(ownerState, breakpoints.keys);
```

Comment:
Isn't it better to append the values you need in the `ownerState`, why create a second arg?